src/ForwardedHttpMiddleware.cs

```diff
@@ -1,27 +1,507 @@
+                        currentValues.RemoteIpAndPort = set.RemoteIpAndPort;
+                        currentValues.ForType = NodeType.Obfuscated;
+                    }
+                    else if (IsUnknownNode(set.RemoteIpAndPortText))
+                    {
+                        applyChanges = true;
+                        set.RemoteIpAndPort = null;
+                        currentValues.RemoteIpAndPortText = set.RemoteIpAndPortText;
+                        currentValues.RemoteIpAndPort = set.RemoteIpAndPort;
+                        currentValues.ForType = NodeType.Unknown;
+                    }
+                    else if (!string.IsNullOrEmpty(set.RemoteIpAndPortText))
+                    {
+                        // Stop at the first unparsable IP, but still apply changes processed so far.
+                        _logger.LogDebug(1, "Unparsable IP: {IpAndPortText}", set.RemoteIpAndPortText);
+                        break;
+                    }
+                }
+                
+                if (checkBy)
+                {
+                    if (IPEndPoint.TryParse(set.LocalIpAndPortText, out var parsedEndPoint))
+                    {
+                        applyChanges = true;
+                        set.LocalIpAndPort = parsedEndPoint;
+                        currentValues.LocalIpAndPortText = set.LocalIpAndPortText;
+                        currentValues.LocalIpAndPort = set.LocalIpAndPort;
+                        currentValues.ForType = NodeType.IP;
+                    }
+                    else if (IsValidObfuscatedNode(set.LocalIpAndPortText))
+                    {
+                        applyChanges = true;
+                        set.RemoteIpAndPort = null;
+                        currentValues.LocalIpAndPortText = set.LocalIpAndPortText;
+                        currentValues.LocalIpAndPort = set.LocalIpAndPort;
+                        currentValues.ForType = NodeType.Obfuscated;
+                    }
+                    else if (IsUnknownNode(set.LocalIpAndPortText))
+                    {
+                        applyChanges = true;
+                        set.RemoteIpAndPort = null;
+                        currentValues.LocalIpAndPortText = set.LocalIpAndPortText;
+                        currentValues.LocalIpAndPort = set.LocalIpAndPort;
+                        currentValues.ForType = NodeType.Unknown;
+                    }
+                    else if (!string.IsNullOrEmpty(set.LocalIpAndPortText))
+                    {
+                        // Stop at the first unparsable IP, but still apply changes processed so far.
+                        _logger.LogDebug(1, "Unparsable IP: {IpAndPortText}", set.LocalIpAndPortText);
+                        break;
+                    }
+                }
+
+                if (checkProto)
+                {
+                    if (!string.IsNullOrEmpty(set.Scheme) && TryValidateScheme(set.Scheme))
```

Comment:
Invalid values shouldn't just be ignored, they should cause the algorithm to fail without applying any of the changes.